src/main/java/com/example/demo/service/UserService.java

```diff
@@ -67,4 +77,55 @@ public User updateProfilePicture(Long id, User user) {
     public void deleteUser(Long id) {
         this.userRepository.deleteById(id);
     }
+
+    public Long findUserIdByUsername(String username) {
+        User user = userRepository.findByUsername(username);
+        if (user != null) {
+            return user.getId();
+        } else {
+            throw new IllegalArgumentException("User not found");
+        }
+    }
+
+    public void addGame(Long userId, Long gameId) {
+        Optional<User> user = userRepository.findById(userId);
```

Comment:
uniformize methodology, 
Use Optional everywhere to manage errors or use methodology above. Select one and stick to it (communicate methodology used to group)